### DISCUSSION TITLE
making survey questions mandatory

### DISCUSSION THREAD

**Discussion by jshinnn** [jspsych/jsPsych#2615] [Q&A] [answered]
Hi, I see that there's a way to make survey questions mandatory for the [survey-text plugin](https://www.jspsych.org/7.0/plugins/survey-text/) with `required: true`. Is there a way to make other survey questions optional / do this with other plugins, specifically the [survey-html-form](https://www.jspsych.org/7.0/plugins/survey-html-form/) plugin? This is in the latest version of jsPsych. Thanks in advance!

**Comment by becky-gilbert** [comment, marked as answer]
Hi @jshinnn, with the survey-html-form plugin you're providing all of the HTML that defines the form elements, so you can make any of them required by adding the [`required` attribute](https://www.w3schools.com/tags/att_input_required.asp) to the input element. Here's an example:
```js
var trial = {
  type: jsPsychSurveyHtmlForm,
  preamble: '<p>How are you feeling <b>right now?</b></p>',
  html: '<p> I am feeling <input name="first" type="text" required />, <input name="second" type="text" required />, and <input name="third" type="text" required />.</p>'
};
```
